### PR TITLE
Improve error message of invalid column refs in insert into

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

Closes https://github.com/crate/crate/issues/12815

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java [hunk 1]
```diff
@@ -1219,6 +1219,13 @@ public void test_insert_returning() {
         printStatement("insert into foo (id, name) values ('string', 1.2) returning foo.*");
     }
 
+    @Test
+    public void test_using_fqn_in_column_list_of_insert_into_results_in_user_friendly_error() {
+        assertThatThrownBy(() -> printStatement("insert into tbl (t.x) values (1)"))
+            .hasMessage("Column references used in INSERT INTO <tbl> (...) must use the column name. They cannot qualify catalog, schema or table. Got `t.x`");
```
Comment (matriv) [inline libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java:1225]: not insisting: Maybe also add the `.isExactlyInstanceOf()`

**Review by matriv — APPROVED** [review]
Thx! Does it deserve a changelog entry?

File: libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java (hunk 1 above)
Comment (mfussenegger) [reply libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java:1225]: The type of exception isn't that important here as no other components rely on it

**Comment by mfussenegger** [comment]
> Does it deserve a changelog entry?

Not sure, I'd tend towards no.

**Pull request merged**